pumps/mongo_aggregate.go

```diff
@@ -249,78 +249,12 @@ func (m *MongoAggregatePump) WriteData(ctx context.Context, data []interface{})
 
 		// put aggregated data into MongoDB
 		for orgID, filteredData := range analyticsPerOrg {
-			collectionName, collErr := m.GetCollectionName(orgID)
-			if collErr != nil {
-				m.log.Info("No OrgID for AnalyticsRecord, skipping")
-				continue
-			}
-
-			thisSession := m.dbSession.Copy()
-			defer thisSession.Close()
-
-			analyticsCollection := thisSession.DB("").C(collectionName)
-			indexCreateErr := m.ensureIndexes(analyticsCollection)
-
-			if indexCreateErr != nil {
-				m.log.Error(indexCreateErr)
-			}
-
-			query := bson.M{
-				"orgid":     filteredData.OrgID,
-				"timestamp": filteredData.TimeStamp,
-			}
-
-			if len(m.dbConf.IgnoreAggregationsList) > 0 {
-				filteredData.DiscardAggregations(m.dbConf.IgnoreAggregationsList)
-			}
-
-			updateDoc := filteredData.AsChange()
-
-			change := mgo.Change{
-				Update:    updateDoc,
-				ReturnNew: true,
-				Upsert:    true,
-			}
-
-			doc := analytics.AnalyticsRecordAggregate{}
-			_, err := analyticsCollection.Find(query).Apply(change, &doc)
-
+			err := m.DoAggregatedWriting(ctx, orgID, filteredData)
 			if err != nil {
-				m.log.WithField("query", query).Error("UPSERT Failure: ", err)
-				return m.HandleWriteErr(err)
+				return err
 			}
 
-			// We have the new doc back, lets fix the averages
-			avgUpdateDoc := doc.AsTimeUpdate()
-			avgChange := mgo.Change{
-				Update:    avgUpdateDoc,
-				ReturnNew: true,
-			}
-			withTimeUpdate := analytics.AnalyticsRecordAggregate{}
-			_, avgErr := analyticsCollection.Find(query).Apply(avgChange, &withTimeUpdate)
-
-			if m.dbConf.ThresholdLenTagList != -1 && (len(withTimeUpdate.Tags) > m.dbConf.ThresholdLenTagList) {
-				m.printAlert(withTimeUpdate, m.dbConf.ThresholdLenTagList)
-			}
-
-			if avgErr != nil {
-				m.log.WithField("query", query).Error("AvgUpdate Failure: ", avgErr)
-				return m.HandleWriteErr(avgErr)
-			}
-
-			m.log.WithFields(logrus.Fields{
-				"collection": collectionName,
-			}).Debug("Wrote aggregated data for ", len(data), " records")
-
-			if m.dbConf.UseMixedCollection {
-				thisData := analytics.AnalyticsRecordAggregate{}
-				err := analyticsCollection.Find(query).One(&thisData)
-				if err != nil {
-					m.log.WithField("query", query).Error("Couldn't find query doc:", err)
-				} else {
-					m.doMixedWrite(thisData, query)
-				}
-			}
+			m.log.Debug("Wrote aggregated data for ", len(data), " records")
```

Comment:
shouldn't it be `len(filteredData)` ?

Reply:
@sredxny nop, since filteredData is the already aggregated records. But I agree that `len(data)` is not correct. 

What do you think if we log `m.log.Debug("Wrote aggregated data for " + orgID)` ?

Reply:
@tbuchaillot that would be better